src/test/run-make/native-link-modifier-verbatim/Makefile

```diff
@@ -0,0 +1,27 @@
+# ignore-cross-compile
+# ignore-macos
+
+include ../../run-make-fulldeps/tools.mk
+
+all:
+	# Verbatim allows specify precise name.
+	$(RUSTC) local_native_dep.rs --crate-type=staticlib -o $(TMPDIR)/local_some_strange_name.ext
+	$(RUSTC) upstream_native_dep.rs --crate-type=staticlib -o $(TMPDIR)/upstream_some_strange_name.ext
+	$(RUSTC) rust_dep.rs -Zunstable-options -l static:+verbatim=upstream_some_strange_name.ext --crate-type rlib
+	$(RUSTC) main.rs --extern rust_dep=$(TMPDIR)/librust_dep.rlib -Zunstable-options -l static:+verbatim=local_some_strange_name.ext
```

Comment:
Could you split this test into two
- in one test `verbatim` is only applied to libraries found by rustc (this case is reliable and cross-platform)
- in another test `verbatim` is applied to libraries found by system linker (this case will need to be ignored on some platforms like macos because not all system linkers support `verbatim`)